site/docs/uk/plugins/guide.md

```diff
@@ -0,0 +1,222 @@
+---
+prev: ../plugins/
+---
+
+# Посібник по плагінам для grammY
+
+Якщо ви бажаєте розробити свій власний плагін та опублікувати його, або якщо ви хочете дізнатися, як працюють плагіни grammY за кулісами, то це місце для вас!
+
+> Зверніть увагу, що вже є опис [плагінів grammY](./) та їх функцій.
+> Ця стаття - це глибоке занурення в їх внутрішні процеси.
+
+## Типи плагінів в grammY
+
+Є два основних типи плагінів в grammY:
+
+- Middleware плагіни: Задачею плагіна є повернути [middleware функцію](../guide/middleware.md), яку можна використовувати в боті grammY.
+- Transformer плагіни: Задачею плагіна є повернути [transformer функцію](../advanced/transformers.md), яку можна використовувати в боті grammY.
+
+Проте іноді ви можете знайти плагіни, які роблять обидві речі.
+Також є інші пакети, які не є функціями middleware або transformer, але ми також називаємо їх плагінами, оскільки вони розширюють функціональність grammY різними способами.
+
+## Правила співпраці
+
+Ви можете опублікувати свій плагін в одному з таких варіантів:
+
+- Опублікувати як **офіційний** плагін.
+- Опублікувати як **плагін від сторонніх розробників**.
+
+Якщо ви вирішите опублікувати свій плагін як сторонній, ми все ще можемо запропонувати вам видне місце на цьому веб-сайті. Однак, ми більш перевагу віддаємо, якщо ви публікуєте свій плагін в [організації grammyjs](https://github.com/grammyjs) на GitHub, зробивши його офіційним плагіном.
+В такому випадку вам буде надано доступ до публікації на GitHub та npm.
+Крім того, ви будете відповідальні за підтримку свого коду.
+
+Перед тим як перейти до деяких практичних прикладів, є кілька правил, на які потрібно звернути увагу, якщо ви хочете, щоб ваші плагіни були включені до списку на цьому веб-сайті.
+
+1. Необхідно мати файл README на GitHub (та npm) з **короткими і чіткими** інструкціями щодо використання плагіна.
+2. Поясніть призначення вашого плагіна та як його використовувати, додавши сторінку до [документації](https://github.com/grammyjs/website). Якщо ви не впевнені, як це зробити, ми можемо створити сторінку за вас.
+3. Оберіть ліцензію з дозволом на використання, наприклад MIT або ISC.
+
+Нарешті, ви повинні знати, що хоча grammY підтримує як Node.js, так і [Deno](https://deno.land/), він є проектом, спрямованим на Deno, і ми також підтримуємо написання плагінів для Deno (і, відповідно, в стилі!).
+Є корисний інструмент під назвою [deno2node](https://github.com/wojpawlik/deno2node), який транспілює ваш код з Deno на Node.js, щоб ми могли однаково добре підтримувати обидві платформи.
+Підтримка Deno є обов'язковою тільки для офіційних плагінів, але не для плагінів сторонніх розробників.
+Тем не менш, ми дуже рекомендуємо спробувати Deno.
```

Comment:
https://onlinecorrector.com.ua/uk/%D0%BE%D0%B4%D0%BD%D0%B0%D0%BA-%D0%BF%D1%80%D0%BE%D1%82%D0%B5-%D0%B0-%D1%82%D0%B0%D0%BA%D0%BE%D0%B6-%D0%B0-%D0%B2%D1%82%D1%96%D0%BC/

```suggestion
Утім, ми дуже рекомендуємо спробувати Deno.
```